dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/icon/IconControllerTest.java

```diff
@@ -60,88 +64,134 @@ void shouldCreateCustomIconWhenFileResourceExist() throws IOException {
-    createCustomIcon(createFileResource(), keywords);
+    createCustomIcon(createFileResource(), keywords, iconKey1);
 
     JsonObject response =
         PUT(
                 "/icons",
                 "{'key':'"
-                    + iconKey
+                    + iconKey1
                     + "', 'description':'"
                     + updatedDescription
                     + "', 'keywords':"
                     + updatedKeywords
                     + "}")
             .content();
 
-    assertEquals(String.format("Icon %s updated", iconKey), response.getString("message").string());
+    assertEquals(
+        String.format("Icon %s updated", iconKey1), response.getString("message").string());
   }
 
   @Test
   void shouldGetOnlyCustomIcons() throws IOException {
     String fileResourceId = createFileResource();
-    createCustomIcon(fileResourceId, keywords);
+    createCustomIcon(fileResourceId, keywords, iconKey1);
+
+    JsonObject content = GET("/icons?keywords=k1").content(HttpStatus.OK);
 
-    JsonList<JsonIcon> icons =
-        GET("/icons?keywords=k1").content(HttpStatus.OK).asList(JsonIcon.class);
+    JsonList<JsonIcon> icons = content.getList("icons", JsonIcon.class);
 
     assertCustomIcons(icons.get(0), keywords, fileResourceId);
   }
 
   @Test
   void shouldGetIconsWithSpecifiedKey() throws IOException {
     String fileResourceId = createFileResource();
-    createCustomIcon(fileResourceId, keyword);
+    createCustomIcon(fileResourceId, keyword, iconKey1);
+
+    JsonObject content = GET("/icons?keywords=m1").content(HttpStatus.OK);
 
-    JsonList<JsonIcon> icons =
-        GET("/icons?keywords=m1").content(HttpStatus.OK).asList(JsonIcon.class);
+    JsonList<JsonIcon> icons = content.getList("icons", JsonIcon.class);
 
     assertCustomIcons(icons.get(0), keyword, fileResourceId);
   }
 
   @Test
   void shouldDeleteIconWhenKeyExists() throws IOException {
-    createCustomIcon(createFileResource(), keywords);
+    createCustomIcon(createFileResource(), keywords, iconKey1);
+
+    JsonObject response = DELETE(String.format("/icons/%s", iconKey1)).content();
+
+    assertEquals(
+        String.format("Icon %s deleted", iconKey1), response.getString("message").string());
+  }
+
+  @Test
+  void shouldGetIconsWithPager() throws IOException {
+
+    int pageSize = 2;
+
+    String fileResourceId1 = createFileResource();
+    createCustomIcon(fileResourceId1, keyword, iconKey1);
+
+    String fileResourceId2 = createFileResource();
+    createCustomIcon(fileResourceId2, keyword, iconKey2);
+
+    String fileResourceId3 = createFileResource();
+    createCustomIcon(fileResourceId3, keyword, iconKey3);
+
+    JsonObject response =
+        GET("/icons?type=CUSTOM&page=1&pageSize=" + pageSize).content(HttpStatus.OK);
+
+    JsonList<JsonIcon> icons = response.getList("icons", JsonIcon.class);
+
+    assertTrue(response.has("pager"));
+    assertEquals(pageSize, icons.size());
```

Comment:
Please follow the tests in https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/ExportControllerPaginationTest.java

You need to
* add tests to show pagination is on by default, maybe get an empty page as well
* assert on pager fields to make sure the numbers are correct
* assert on fields which should not be present like `total` and `pageCount`

Reply:
Added